src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -199,65 +201,52 @@ else if (address != null) {
-            for (KitRequest kit : kitRequests) {
-                if (!orderNumbers.contains(kit.getExternalOrderNumber()) && StringUtils.isNotBlank(kit.getExternalOrderNumber())) {
-                    orderNumbers.add(kit.getExternalOrderNumber());
-                    externalOrdersStatus.put(kit.getExternalOrderNumber(), kit);
-                }
-            }
-            JSONObject payload = new JSONObject().put("orderNumbers", orderNumbers);
-            logger.info("payload size for order status: " + orderNumbers.size());
-            String sendRequest = DSMServer.getBaseUrl(getExternalShipperName()) + STATUS_ENDPOINT;
-            logger.info("payload: " + payload.toString());
-            Response gbfResponse = executePost(Response.class, sendRequest, payload.toString(), DSMServer.getApiKey(getExternalShipperName()));
-            logger.info(gbfResponse.getXML());
-            if (gbfResponse != null && gbfResponse.isSuccess()) {
-                List<Status> statuses = gbfResponse.getStatuses();
-                logger.info("Got a list of " + statuses.size() + " responses from GBF status");
-                if (statuses != null && !statuses.isEmpty()) {
-                    for (Status status : statuses) {
-                        logger.info("Got Kit status in GBF Response is " + status.getOrderStatus() + " for " + status.getOrderNumber());
-                        KitRequest kit = externalOrdersStatus.get(status.getOrderNumber());
-                        if (!status.getOrderStatus().equals(kit.getExternalOrderStatus())) {
-                            logger.info("Kit status from GBF is changed from " + kit.getExternalOrderStatus() + " to " + status.getOrderStatus() + " for kit: " + kit.getExternalOrderNumber());
-                        }
-                        KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_SENT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER, kit.getExternalOrderNumber(), 2);//todo change this to the number of subkits but for now 2 for test boston works
-                        if (status.getOrderStatus().equals(NOT_FOUND) && StringUtils.isNotBlank(kit.getExternalOrderStatus()) && kit.getExternalOrderStatus().equals("NOT FOUND")
-                                && System.currentTimeMillis() - kit.getExternalOrderDate() >= TimeUnit.HOURS.toMillis(24)) {
-                            List<String> dsmKitRequestIds = getDSMKitRequestIds(status.getOrderNumber());
-                            if (dsmKitRequestIds != null && !dsmKitRequestIds.isEmpty()) {
-                                for (String dsmKitRequestId : dsmKitRequestIds) {
-                                    KitRequestExternal.updateKitRequest(status.getOrderStatus(), System.currentTimeMillis(), dsmKitRequestId);// in order to update time for the  next 24 hour check we need this
-                                }
+    public void orderStatus(Connection conn, KitRequest kit) throws Exception {
```

Comment:
Rather than iterating on a list, this now operates on a single kit at a time, using the passed-in `Connection`.